File: ainpc-core-plugin/src/main/java/ro/ainpc/gui/GuiSessionManager.java
package ro.ainpc.gui;

import org.bukkit.entity.Player;

import java.util.Optional;
import java.util.UUID;
import java.util.concurrent.ConcurrentHashMap;
import java.util.concurrent.ConcurrentMap;

public class GuiSessionManager {

    private final ConcurrentMap<UUID, GuiSession> sessionsById = new ConcurrentHashMap<>();
    private final ConcurrentMap<UUID, UUID> sessionsByPlayer = new ConcurrentHashMap<>();

    public GuiSession create(Player player, GuiKey key) {
        closePlayer(player.getUniqueId());

        GuiSession session = new GuiSession(
            UUID.randomUUID(),
            player.getUniqueId(),
            key,
            System.currentTimeMillis()
        );
        sessionsById.put(session.getSessionId(), session);
        sessionsByPlayer.put(player.getUniqueId(), session.getSessionId());
        return session;
    }

    public Optional<GuiSession> find(UUID sessionId) {
        return Optional.ofNullable(sessionsById.get(sessionId));
    }

    public void close(UUID sessionId) {
        GuiSession removed = sessionsById.remove(sessionId);
        if (removed != null) {
            sessionsByPlayer.remove(removed.getPlayerId(), sessionId);
        }
    }

    public void closePlayer(UUID playerId) {
        UUID sessionId = sessionsByPlayer.remove(playerId);
        if (sessionId != null) {
            sessionsById.remove(sessionId);
        }
    }

    public void closeAll() {
        sessionsById.clear();
        sessionsByPlayer.clear();
    }
}

File: ainpc-core-plugin/bin/main/ro/ainpc/gui/GuiSessionManager.kt
package ro.ainpc.gui

import org.bukkit.entity.Player
import java.util.Optional
import java.util.UUID
import java.util.concurrent.ConcurrentHashMap
import java.util.concurrent.ConcurrentMap

class GuiSessionManager {
    private val sessionsById: ConcurrentMap<UUID, GuiSession> = ConcurrentHashMap()
    private val sessionsByPlayer: ConcurrentMap<UUID, UUID> = ConcurrentHashMap()

    fun create(player: Player, key: GuiKey): GuiSession {
        closePlayer(player.uniqueId)

        val session = GuiSession(
            UUID.randomUUID(),
            player.uniqueId,
            key,
            System.currentTimeMillis()
        )
        sessionsById[session.getSessionId()] = session
        sessionsByPlayer[player.uniqueId] = session.getSessionId()
        return session
    }

    fun find(sessionId: UUID): Optional<GuiSession> = Optional.ofNullable(sessionsById[sessionId])

    fun close(sessionId: UUID) {
        val removed = sessionsById.remove(sessionId)
        if (removed != null) {
            sessionsByPlayer.remove(removed.getPlayerId(), sessionId)
        }
    }

    fun closePlayer(playerId: UUID) {
        val sessionId = sessionsByPlayer.remove(playerId)
        if (sessionId != null) {
            sessionsById.remove(sessionId)
        }
    }

    fun closeAll() {
        sessionsById.clear()
        sessionsByPlayer.clear()
    }
}
